superset/assets/javascripts/explore/components/controls/FixedOrMetricControl.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Label, Popover, OverlayTrigger } from 'react-bootstrap';
+
+import controls from '../../stores/controls';
+import TextControl from './TextControl';
+import SelectControl from './SelectControl';
+import ControlHeader from '../ControlHeader';
+import PopoverSection from '../../../components/PopoverSection';
+
+const propTypes = {
+  onChange: PropTypes.func,
+  value: PropTypes.object,
+  isFloat: PropTypes.bool,
+  datasource: PropTypes.object,
+  default: PropTypes.object,
+};
+
+const defaultProps = {
+  onChange: () => {},
+  default: { type: 'fix', value: 5 },
+};
+
+export default class FixedOrMetricControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = this.onChange.bind(this);
+    const type = props.value ? props.value.type : props.default.type || 'fix';
+    const value = props.value ? props.value.value : props.default.value || '100';
+    this.state = {
+      type,
+      fixedValue: type === 'fix' ? value : '',
+      metricValue: type === 'metric' ? value : null,
```

Comment:
`VALUE_TYPE_METRIC` instead of `metric`? Also it looks weird that one of them  defaults to `''` while the other one defaults to `null`

Reply:
one is a string (TextControl) and the other is an unselected SelectControl...